src/UnitTests/test_quote_prov.cpp

```diff
@@ -33,50 +42,99 @@ static sgx_ql_set_logging_function_t sgx_ql_set_logging_function;
 
 static void Log(sgx_ql_log_level_t level, const char* message)
 {
-    char const * levelText = "ERROR";
-    switch (level) {
+    char const* levelText = "ERROR";
+    switch (level)
+    {
         case SGX_QL_LOG_WARNING:
             levelText = "WARNING";
             break;
         case SGX_QL_LOG_INFO:
             levelText = "INFO";
             break;
         default:
-	    levelText = "ERROR";
+            levelText = "ERROR";
     }
-    printf(
-        "[%s]: %s\n",
-        levelText,
-        message);
+    printf("[%s]: %s\n", levelText, message);
 }
 
+#if defined __LINUX__
 static void* LoadFunctions()
 {
-    // if this assert fails, be sure libdcap_quoteprov.so is in your LD_LIBRARY_PATH
+    // if this assert fails, be sure libdcap_quoteprov.so is in your
+    // LD_LIBRARY_PATH
+
     void* library = dlopen("libdcap_quoteprov.so", RTLD_NOW);
+    printf(dlerror());
     if (library == nullptr)
     {
         Log(SGX_QL_LOG_ERROR, dlerror());
         abort();
     }
 
-    sgx_ql_free_revocation_info = reinterpret_cast<sgx_ql_free_revocation_info_t>(dlsym(library, "sgx_ql_free_revocation_info"));
+    sgx_ql_free_revocation_info =
+        reinterpret_cast<sgx_ql_free_revocation_info_t>(
+            dlsym(library, "sgx_ql_free_revocation_info"));
     assert(sgx_ql_free_revocation_info);
 
-    sgx_ql_get_revocation_info = reinterpret_cast<sgx_ql_get_revocation_info_t>(dlsym(library, "sgx_ql_get_revocation_info"));
+    sgx_ql_get_revocation_info = reinterpret_cast<sgx_ql_get_revocation_info_t>(
+        dlsym(library, "sgx_ql_get_revocation_info"));

```

Comment:
Why not isolate the GetProcAddress/dlsym functions into a common function which implements the functionality - that way the Linux and Windows code which does the loading from the shared object can be the same.